bin/i18n/redact_restore_utils.rb

```diff
@@ -57,70 +60,54 @@ def self.restore_data(source_data, redacted_data, plugins = [], format = 'md')
     source_json.close
     redacted_json.close
 
-    return restored
+    return JSON.parse(restored)
   end
 
   def self.restore(source, redacted, dest, plugins = [], format = 'md')
     return unless File.exist?(source)
     return unless File.exist?(redacted)
-    is_json = File.extname(source) == '.json'
-    source_data =
-      if is_json
-        JSON.parse(File.read(source))
-      else
-        YAML.load_file(source)
-      end
-    redacted_data =
-      if is_json
-        JSON.parse(File.read(redacted))
-      else
-        YAML.load_file(redacted)
-      end
 
-    return unless source_data
-    return unless redacted_data
-    return unless source_data&.values&.first&.length
-    return unless redacted_data&.values&.first&.length
+    if I18nScriptUtils.json_file?(source) || I18nScriptUtils.yaml_file?(source)
+      source_data = I18nScriptUtils.parse_file(source)
+      redacted_data = I18nScriptUtils.parse_file(redacted)
 
-    restored =
-      if is_json
-        RedactRestoreUtils.restore_data(source_data, redacted_data, plugins, format)
-      else
-        RedactRestoreUtils.restore_data(source_data.values.first, redacted_data.values.first, plugins, format)
-      end
+      return unless source_data
+      return unless redacted_data
+      return unless source_data&.values&.first&.length
+      return unless redacted_data&.values&.first&.length
 
-    File.open(dest, "w+") do |file|
-      if File.extname(dest) == '.json'
-        file.write(JSON.pretty_generate(restored))
-      else
+      # yaml files have a locale key wrapping all content. Redacted files have the translated language key, while the
+      # un-redacted file keeps the English key. We need to extract the data inside the locale key to restore content.
```

Comment:
This condition complicates and limits the functionality of this method because it will work correctly only with one specific YAML file structure.

Reply:
It would be better to create a separate method just for Markdown files with a specific format.